client/snapshotter_opts_unix.go

```diff
@@ -110,7 +111,11 @@ func resolveSnapshotOptions(ctx context.Context, client *Client, snapshotterName
 		return "", err
 	}
 	// TODO(dgl): length isn't taken into account here yet either.
```

Comment:
Since you're already touching this, does it make sense to update the snapshot naming here and in `container_opts_unix.go` to encode the full idmap rather than just the root user map? 

I'm not sure how likely it would be, but you could get the wrong result if two configs on the same host have the same root mapping, but different non-root mappings.

Reply:
The updated version will use the hash of (parent, idmap) for the snapshot name. That way should address this issue.